Data/Yaml/Builder.hs

```diff
@@ -88,8 +89,13 @@ string s   =
         | otherwise = EventScalar (encodeUtf8 s) StrTag PlainNoTag Nothing
  
 -- Use aeson's implementation which gets rid of annoying decimal points
+#if MIN_VERSION_aeson(0, 7, 0)
 scientific :: Scientific -> YamlBuilder
 scientific n = YamlBuilder (EventScalar (TE.encodeUtf8 $ TL.toStrict $ toLazyText $ encodeToTextBuilder (Number n)) IntTag PlainNoTag Nothing :)
```

Comment:
Things excluded from imports by `#if MIN_VERSION_aeson(0, 7, 0)` thus leading to compilation error unless also wrapped into `#if MIN_VERSION_aeson(0, 7, 0)`.